### PR TITLE
fix: improved team upgrade screen to also show unpublished teams

### LINKED ISSUE #12468: [CAL-2743] show unpublished teams
right now, if you have an unpublished team, it still shows the empty screen

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2743](https://linear.app/calcom/issue/CAL-2743/show-unpublished-teams)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
todo:

- [x] figure out hasUnpublishedTeam

before:

- even if you have an unpublished team, you see the upgrade screen:

CleanShot 2023-11-22 at 13 29 12@2x

after:
CleanShot 2023-11-22 at 14 03 05@2x

### CODE REVIEW

File: packages/features/ee/teams/components/TeamsListing.tsx [hunk 1]
```diff
@@ -102,7 +104,7 @@ export function TeamsListing() {
           <TeamList teams={invites} pending />
         </div>
       )}
-
+      {/* TODO: Also show teams that have no slug here, so we can show unpublished teams */}
```
Comment (PeerRich) [inline packages/features/ee/teams/components/TeamsListing.tsx:107]:
```suggestion
```

File: packages/features/tips/UpgradeTip.tsx [hunk 2]
```diff
@@ -32,7 +32,10 @@ export function UpgradeTip({
   const hasEnterprisePlan = false;
   //const { isLoading , hasEnterprisePlan } = useHasEnterprisePlan();
 
-  if (plan === "team" && hasTeamPlan) return children;
+  // figure out how to check for unpublished team (idea: fetch all teams and check for !team.slug)
+  const hasUnpublishedTeam = true;
```
Comment (PeerRich) [inline packages/features/tips/UpgradeTip.tsx:36]: todo: figure out how to check for unpublished team (idea: fetch all teams and check for !team.slug)

File: packages/features/ee/teams/components/TeamsListing.tsx [hunk 3]
```diff
@@ -92,6 +92,8 @@ export function TeamsListing() {
     return <SkeletonLoaderTeamList />;
   }
 
+  console.log(teams.length > 0);
+
```
Comment (zomars) [inline packages/features/ee/teams/components/TeamsListing.tsx:96]:
```suggestion
```

File: packages/trpc/server/routers/viewer/teams/_router.tsx
Comment (zomars) [inline packages/trpc/server/routers/viewer/teams/_router.tsx:1]: Reduced team router boilerplate

**Review by zomars — DISMISSED** [review]

**Review by zomars — APPROVED** [review]

**Pull request merged**